adafruit_fram.py

```diff
@@ -93,103 +108,58 @@ def write_protected(self):
             When no ``WP`` pin is supplied, protection is only at the software
             level in this library.
         """
-        if not self._wp_pin is None:
-            status = self._wp_pin.value
-        else:
-            status = self._wp
-        return status
+        return self._wp if self._wp_pin is None else self._wp_pin.value
 
     @write_protected.setter
     def write_protected(self, value):
         self._wp = value
         if not self._wp_pin is None:
             self._wp_pin.value = value
 
-    def write_protect_pin(self, wp_pin, write_protect=False):
-        """ Assigns the write protection (``WP``) pin.
+    def __getitem__(self, key):
+        if isinstance(key, int):
+            if key > self._max_size:
+                raise ValueError("Register '{0}' greater than maximum FRAM size."
+                                 " ({1})".format(key, self._max_size))
+            return self._read_byte(key)
```

Comment:
Maybe have this be:
```python
return self._read_byte(key)[0]
```
so you just get back the value instead of a single element bytearray with the value.

Reply:
Meant to comment on this last night. With the change to `__getitem__`, I wanted to make the return value standard. I'm not completely opposed to sending back the value. Just felt that reduced API confusion was more beneficial.